core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -39,6 +39,12 @@ import kafka.server.KafkaServer
 /* We have some tests in this class instead of `BaseConsumerTest` in order to keep the build time under control. */
 class PlaintextConsumerTest extends BaseConsumerTest {
 
+  @Before
+  override def setUp() {
+    super.setUp
+    this.consumerConfig.remove(ConsumerConfig.CLIENT_ID_CONFIG)
```

Comment:
To me, it seems to be easier to remove the config and rely on `KafkaConsumer`'s internal auto ID assignment that uses incremental counter and thus generates new unique ID automatically -- otherwise, even if a test does not care about the ID but uses multiple consumers with shared config it needs to change consumer ID.

But we can change if you thinks this code is bad test setup.